### ISSUE TITLE
[Accessibility]: No indication of newly loaded content or content loading for slow network fetches

### ISSUE THREAD

**Issue by Strum355** [sourcegraph/sourcegraph#35318]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33515

### Problem description

When loading the autoindexing jobs list, it can take potentially double digit seconds to load the actual list due to the query size. There is no announcement of content having loaded or that content is in the process of being loaded without traveling all the way to the loading icon.

![image](https://user-images.githubusercontent.com/18282288/167929309-bb1eb6ec-16a0-41bb-9437-d7a9d17b8638.png)

### Expected behavior

Announce when content has been loaded and better indicate that something is still loading.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by umpox** [comment]
I'm looking into this, I think a suitable fix would be to add `aria-live="polite"` to `LoadingSpinner`. Any other announcements might be overkill, as otherwise we might find ourselves overly notifying on the page for different network requests.
